platforms/jvm/code-quality/src/main/java/org/gradle/api/plugins/quality/internal/CodeNarcInvoker.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.gradle.api.plugins.quality.internal;
+
+import com.google.common.collect.ImmutableMap;
+import org.gradle.api.Action;
+import org.gradle.api.GradleException;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.file.RegularFile;
+import org.gradle.api.internal.exceptions.MarkedVerificationException;
+import org.gradle.api.internal.project.ant.AntLoggingAdapter;
+import org.gradle.api.internal.project.antbuilder.AntBuilderDelegate;
+import org.gradle.internal.logging.ConsoleRenderer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.gradle.api.plugins.quality.internal.AntInvokeUtils.invoke;
+
+class CodeNarcInvoker implements Action<AntBuilderDelegate> {
```

Comment:
💭 This is just rewritten from Groovy to Java so we can use annotation processor for code-quality project